### PR TITLE
Formalize the use of bounded integers and error-on-overflow semantics

### PR DESCRIPTION
**Pull request by emina**
This PR changes the formalization of the Cedar semantics to reflect the fact that we're using signed 64-bit integers, along with error-on-overflow semantics for arithmetic operators.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aaronjeline** [comment]
This PR should either include or create a new issue to change the DRT to not ignore mismatches on integer overflow errors

**Review by andrewmwells-amazon — APPROVED** [review]

**Comment by khieta** [comment]
> This PR should either include or create a new issue to change the DRT to not ignore mismatches on integer overflow errors

Not needed, since the Lean formalization isn't plugged into DRT yet. We should make sure this is done in #30 though.

**Review by khieta — APPROVED** [review]
Lgtm.

File: cedar-lean/Cedar/Data/Int64.lean [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+instance : Inhabited Int64 where
+  default := Subtype.mk 0 (by simp)
+
+
+namespace Int64
+
+----- Definitions -----
+
+def mk (i : Int) (h : INT64_MIN ≤ i ∧ i ≤ INT64_MAX) : Int64 :=
```
Comment (shaobo-he-aws) [inline cedar-lean/Cedar/Data/Int64.lean:38]: Nit: is it possible to put `INT64_MIN ≤ i ∧ i ≤ INT64_MAX` in a predicate (function)?

**Review by shaobo-he-aws — APPROVED** [review]

**Pull request merged**